### PR TITLE
Difficulty adjustment algorithm fix, updated total mone supply for 2:1 zcl

### PR DESCRIPTION
**Pull request by kaypon**

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -191,33 +190,30 @@ class CTestNetParams : public CMainParams {
...
+        vSeeds.push_back(CDNSSeedData("anon1-testnet", "198.58.103.84"));
+        vSeeds.push_back(CDNSSeedData("anon2-testnet", "50.116.27.226"));
+        vSeeds.push_back(CDNSSeedData("anon3-testnet", "198.58.97.186"));
 
         // guarantees the first 2 characters, when base58 encoded, are "tA"
         base58Prefixes[PUBKEY_ADDRESS]     = {0x1C,0xCE};
         // guarantees the first 2 characters, when base58 encoded, are "t2"
         base58Prefixes[SCRIPT_ADDRESS]     = {0x1C,0xBA};
         // the first character, when base58 encoded, is "9" or "c" (as in Bitcoin)
-        base58Prefixes[SECRET_KEY]         = {0xEF};
```
Comment (nlevo) [inline src/chainparams.cpp]: Keep bitcoin testnet private key prefix {0xEF};

File: README.md [hunk 2]
```diff
@@ -27,6 +27,12 @@ Build
 -----------------
 ### Linux
 
+Update System:
+```{r, engine='bash'}
+sudo apt-get update
+sudo apt-get upgrade
+```
```
Comment (thedon-chris) [inline README.md]: Why is this here? Reference Issue #27 

Please keep code that belongs to issues in separate PR's and don't mix code that belongs to a feature with issue fixes

File: src/amount.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ static const CAmount CENT = 1000000;
  * critical; in unusual circumstances like a(nother) overflow bug that allowed
  * for the creation of coins out of thin air modification could lead to a fork.
  * */
-static const CAmount MAX_MONEY = 31500000 * COIN;
+static const CAmount MAX_MONEY = 40170000 * COIN;
```
Comment (thedon-chris) [inline src/amount.h]: Please edit PR comments to include money change supply

File: src/chainparams.cpp [hunk 4]
```diff
@@ -34,20 +34,19 @@ class CMainParams : public CChainParams {
...
+        consensus.nSubsidySlowStartInterval = 1;
+        consensus.nSubsidyHalvingInterval = 134000;
         consensus.nMajorityEnforceBlockUpgrade = 750;
         consensus.nMajorityRejectBlockOutdated = 950;
         consensus.nMajorityWindow = 4000;
         consensus.prePowLimit = uint256S("0007ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
         consensus.powLimit = uint256S("07ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
-        consensus.nPowAveragingWindow = 2016; //diffuculty adjusts every 2 weeks
-        // assert(maxUint/UintToArith256(consensus.powLimit) >= consensus.nPowAveragingWindow);
+        consensus.nPowAveragingWindow = 17; //diffuculty adjusts every 2 weeks
```
Comment (thedon-chris) [inline src/chainparams.cpp]: Please add this to commit description

File: src/miner.cpp [hunk 5]
```diff
@@ -218,7 +218,7 @@ CBlockTemplate* CreateNewForkBlock(bool& bFileNotFound, const int nHeight)
 
 
     // Largest block you're willing to create:
-    unsigned int nBlockMaxSize = (unsigned int)(MAX_BLOCK_SIZE) + 4000000;
+    unsigned int nBlockMaxSize = (unsigned int)(MAX_BLOCK_SIZE) + 1000000;
```
Comment (thedon-chris) [inline src/miner.cpp]: This will cause an error during Mainnet launch of larger fork blocks

File: src/pow.cpp [hunk 6]
```diff
@@ -38,7 +39,7 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
         return nProofOfWorkLimit;
 
     // right at fork
-    else if(isForkBlock(nHeight) && !isForkBlock(nHeight - params.nPowAveragingWindow))
```
Comment (thedon-chris) [inline src/pow.cpp]: Why has this been removed?

File: src/miner.cpp (hunk 5 above)
Comment (nlevo) [reply src/miner.cpp]: No, it shouldn't. All fork blocks will be under 1MB.

File: src/pow.cpp (hunk 6 above)
Comment (nlevo) [reply src/pow.cpp]: It was removed to keep difficulty constant during fork period.

File: README.md (hunk 2 above)
Comment (kaypon) [reply README.md]: Will do in the future.
